### ISSUE TITLE
We should support provisioning of smaller environment for Integration CI purposes

### ISSUE THREAD

**Issue by etsauer** [rhtconsulting/rhc-ose#42]
Currently we require new Environments to have nodes. Because of a limitation of the nova `--num-instances` flag, our scripts currently require a minimum of 2 nodes, other than the master. That means that our minimal install of openshift uses up 3 m1.large instances. I think we can improve upon this in two ways:
- Enhance the scripts to allow for an install of fewer additional nodes (0 or 1)
- Enable our scripts to work with smaller instance flavors (lower memory footprint)

/cc @oybed @timothyh

**Comment by etsauer** [comment]
@oybed was brainstorming some ideas for this one last night... one road bump we are going to come across is what to do with DNS when we do an all-in-one install... thoughts?

**Comment by oybed** [comment]
@etsauer yes, very true - since SkyDNS takes up 53 (both udp and tcp). The alternatives we have are:
- require an external DNS server
- stand up a very small DNS server in addition to the all-in-one
- see if we can in some way tie in with SkyDNS
- run our own DNS server (e.g.: bind) in parallel with SkyDNS but using a different port. This makes for an ugly scenario, but we may be able to do something with iptables to route requests to it. I'd consider this to be the very last resort solution.

BTW: even for the "quick installation" as outlined in the docs, being an all-in-one, I believe they require an external DNS server.
